open-ce/utils.py

```diff
@@ -150,3 +152,39 @@ def get_output(command):
     '''Print and execute a shell command and then return the output.'''
     print("--->{}".format(command))
     return subprocess.check_output(command, shell=True).decode("utf-8").strip()
+
+def variant_key(py_ver, build_type):
+    '''
+    Returns a variant key using python version and build type
+    '''
+    result = ""
+    if py_ver:
+        result +=  "py" + py_ver
+    if build_type:
+        result +=  "-" + build_type
+    return result
+
+def generalize_version(package):
+    """Add `.*` to package versions when it is needed."""
+
+    # Remove multiple spaces or tabs
+    package = re.sub(' +', ' ', package)
+
+    # Check if we want to add .* to the end of versions
+    py_matched = re.match(r'(\w+[-]*\w+)([\s,=,<,>]*)(.*)', package)
```

Comment:
These may be better regexes to use.
```suggestion
    package = re.sub('\s+', ' ', package)

    # Check if we want to add .* to the end of versions
    py_matched = re.match(r'([\w-]+)([\s=<>]*)(.*)', package)
```